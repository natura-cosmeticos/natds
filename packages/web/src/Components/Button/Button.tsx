import * as React from 'react'

import MaterialButton from '@material-ui/core/Button'
import { IButtonProps } from './Button.props'
import useStyles from './Button.styles'

export { IButtonProps } from './Button.props'

/**
 * ```
 * import { Button } from '@naturacosmeticos/natds-web';
 * ```
 *
 * For more advanced usages, check [Material UI Button docs](https://material-ui.com/components/buttons/)
 */
export const Button = React.forwardRef<HTMLButtonElement, IButtonProps>(
  (props: IButtonProps, ref) => {
    const { classes, disableElevation, ...otherProps } = props

    const mergedClasses = {
      ...classes,
      ...useStyles()
    }

    return (
      <MaterialButton
        classes={mergedClasses}
        disableElevation={disableElevation}
        ref={ref}
        {...otherProps}
      />
    )
  }
)

Button.displayName = 'Button'
export default Button
