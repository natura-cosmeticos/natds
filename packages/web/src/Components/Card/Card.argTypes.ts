/* eslint-disable no-magic-numbers */
export const argTypes = {
  children: {
    description: 'The content of the component.',
    table: { type: { summary: 'node' } },
    control: { type: null }
  },
  className: {
    control: { type: null }
  },
  elevation: {
    control: {
      type: 'inline-radio',
      options: [0, 1]
    },
    description: 'Shadow depth, corresponds to dp in the spec.',
    table: {
      type: { summary: 'number' }
    }
  },
  square: {
    description: 'If `true`, rounded corners are disabled.',
    table: {
      defaultValue: { summary: false },
      type: { summary: 'bool' }
    }
  }
}

export default argTypes
