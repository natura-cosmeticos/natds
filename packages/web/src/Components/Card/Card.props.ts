/* eslint-disable no-magic-numbers */
import { PaperProps } from "@material-ui/core";

export interface ICardProps
  extends Pick<PaperProps, "children" | "elevation" | "square"> {

  /**
   * The content of the component.
   *
   */
  children: PaperProps["children"];

  /**
   * Override or extend the styles applied to the component
   */
  className: PaperProps["className"];

  /**
   * If `true`, no elevation is used.
   *
   * @default 1
   * @optional
   * @type number
   */
  elevation?: PaperProps["elevation"];

  /**
   * If `true`, rounded corners are disabled.
   * @default false
   * @optional
   * @type boolean
   */
  square?: PaperProps["square"];
}
