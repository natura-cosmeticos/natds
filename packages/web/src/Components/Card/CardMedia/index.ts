import CardMedia from "./CardMedia";

export { ICardMediaProps } from "./CardMedia.props";
export { CardMedia };
export default CardMedia;
