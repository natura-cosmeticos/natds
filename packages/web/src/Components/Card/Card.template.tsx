import * as React from "react";
import { makeStyles, createStyles } from "@material-ui/core";
import { Story } from "@storybook/react";
import Button from "../Button";
import Typography from "../Typography";
import { Card, ICardProps } from "./index";

const useStyles = makeStyles(
  createStyles({
    card: {
      padding: 16,
      maxWidth: 328,
    },
    body: {
      margin: "16px 0",
    },
  }),
  { name: "CardDemo" }
);

const CardTemplate: Story<ICardProps> = (args: ICardProps) => {
  const classes = useStyles();

  return (
    <Card {...args} className={classes.card}>
      <Typography variant="overline">overline</Typography>
      <Typography variant="h6">Header 6</Typography>
      <Typography variant="caption">Caption</Typography>
      <Typography variant="body2" className={classes.body}>
        This is an example of text using the body2 variant of Typography
        component
      </Typography>
      <Button variant="contained" color="primary">
        Default
      </Button>
    </Card>
  );
};

export default CardTemplate;
