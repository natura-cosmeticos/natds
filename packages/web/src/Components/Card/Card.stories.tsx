import { Card } from "./Card";
import { argTypes } from "./Card.argTypes";
import CardTemplate from "./Card.template";

export default {
  argTypes,
  component: Card,
  title: "Components/Card",
};

export const Playground = CardTemplate.bind({});