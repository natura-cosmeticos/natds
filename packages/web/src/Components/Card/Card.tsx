/* eslint-disable no-magic-numbers */
import * as React from "react";
import MaterialPaper from "@material-ui/core/Paper";
import { ICardProps } from "./Card.props";

/**
 *
 * ```
 * import { Card } from "@naturacosmeticos/natds-web";
 * ```
 *
 */

export const Card = (props: ICardProps, ref) => (<MaterialPaper {...props} ref={ref} />);

Card.displayName = "Card";

/**
 * @deprecated
 */

export const Paper = Card;

export default { Card, Paper };
