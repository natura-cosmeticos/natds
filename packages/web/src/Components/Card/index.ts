import Card from "./Card";

export { ICardProps } from "./Card.props";
export { Card };
export default Card;
