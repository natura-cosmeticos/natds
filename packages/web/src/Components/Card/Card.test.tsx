import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import { Card } from './Card'
import { Button } from '../Button'

describe('Card components', () => {
  test('should match to snapshot - default props', () => {
    const testRenderer = TestRenderer.create(<Card />)

    expect(testRenderer).toMatchSnapshot('Card empty snapshot')
  })
  test('should match to snapshot - with elevation 0', () => {
    const testRenderer = TestRenderer.create(<Card elevation={0} />)

    expect(testRenderer).toMatchSnapshot('Card empty snapshot')
  })

  test('should match to snapshot - with square true', () => {
    const testRenderer = TestRenderer.create(<Card square />)

    expect(testRenderer).toMatchSnapshot('Card empty snapshot')
  })

  test('should match to snapshot - with children', () => {
    const testRenderer = TestRenderer.create(
      <Card>
        <Button>default</Button>
      </Card>
    )

    expect(testRenderer).toMatchSnapshot('Card empty snapshot')
  })
})
