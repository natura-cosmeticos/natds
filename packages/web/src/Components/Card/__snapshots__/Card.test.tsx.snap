// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card components should match to snapshot - default props: Card empty snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
/>
`;

exports[`Card components should match to snapshot - with children: Card empty snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiButton-root NatDSButton-root-1 MuiButton-text"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label NatDSButton-label-2"
    >
      default
    </span>
  </button>
</div>
`;

exports[`Card components should match to snapshot - without elevation: Card empty snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation0 MuiPaper-rounded"
/>
`;

exports[`Card components should match to snapshot - without square: Card empty snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1"
/>
`;
